frame/nis/src/lib.rs

```diff
@@ -0,0 +1,937 @@
+		/// Information on runtime weights.
+		type WeightInfo: WeightInfo;
+
+		/// Overarching event type.
+		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+
+		/// The treasury's pallet id, used for deriving its sovereign account ID.
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+
+		/// Currency type that this works on.
+		type Currency: ReservableCurrency<Self::AccountId, Balance = Self::CurrencyBalance>;
+
+		/// Just the `Currency::Balance` type; we have this item to allow us to constrain it to
+		/// `From<u64>`.
+		type CurrencyBalance: sp_runtime::traits::AtLeast32BitUnsigned
+			+ codec::FullCodec
+			+ Copy
+			+ MaybeSerializeDeserialize
+			+ sp_std::fmt::Debug
+			+ Default
+			+ From<u64>
+			+ TypeInfo
+			+ MaxEncodedLen;
+
+		/// Origin required for auto-funding the deficit.
+		type FundOrigin: EnsureOrigin<Self::RuntimeOrigin>;
+
+		/// The issuance to ignore. This is subtracted from the `Currency`'s `total_issuance` to get
+		/// the issuance with which we determine the thawed value of a given proportion.
+		type IgnoredIssuance: Get<BalanceOf<Self>>;
+
+		/// The accounting system for the fungible counterpart tokens.
+		type Counterpart: FungibleMutate<Self::AccountId>;
+
+		/// The system to convert an overall proportion of issuance into a number of fungible
+		/// counterpart tokens.
+		///
+		/// In general it's best to use `WithMaximumOf`.
+		type CounterpartAmount: ConvertBack<
+			Perquintill,
+			<Self::Counterpart as FungibleInspect<Self::AccountId>>::Balance,
+		>;
+
+		/// Unbalanced handler to account for funds created (in case of a higher total issuance over
+		/// freezing period).
+		type Deficit: OnUnbalanced<PositiveImbalanceOf<Self>>;
+
+		/// The target sum of all receipts' proportions.
+		type Target: Get<Perquintill>;
+
+		/// Number of duration queues in total. This sets the maximum duration supported, which is
+		/// this value multiplied by `Period`.
+		#[pallet::constant]
+		type QueueCount: Get<u32>;
+
+		/// Maximum number of items that may be in each duration queue.
+		///
+		/// Must be larger than zero.
+		#[pallet::constant]
+		type MaxQueueLen: Get<u32>;
+
+		/// Portion of the queue which is free from ordering and just a FIFO.
+		///
+		/// Must be no greater than `MaxQueueLen`.
+		#[pallet::constant]
+		type FifoQueueLen: Get<u32>;
+
+		/// The base period for the duration queues. This is the common multiple across all
+		/// supported freezing durations that can be bid upon.
+		#[pallet::constant]
+		type Period: Get<Self::BlockNumber>;
+
+		/// The minimum amount of funds that may be placed in a bid. Note that this
+		/// does not actually limit the amount which may be represented in a receipt since bids may
+		/// be split up by the system.
+		///
+		/// It should be at least big enough to ensure that there is no possible storage spam attack
+		/// or queue-filling attack.
+		#[pallet::constant]
+		type MinBid: Get<BalanceOf<Self>>;
```

Comment:
2 cents: over time, I've started to lean more and more toward having these configurations as storage items rather than values that can only be changed via runtime upgrade only. 

It is slightly slower, but we're also far far from being at a point where that extra read is causing anyone a problem, but needing to sync parameter updates with the release is a problem.